src/g42-main.cpp

```diff
@@ -339,7 +339,9 @@ int main(int argc, const char **argv) { // the main() function
 	g_dbg_level = 60;
 	if (early_debug) g_dbg_level_set(20, mo_file_reader::gettext("L_early_debug_comand_line"));
 
-	my_cap::drop_root(); // [SECURITY] if we are started as root, then here drop the UID/GID (we retain CAPs).
+	std::string home_dir = my_cap::drop_root(!remove_and_count(argt, "--home-env" )); // [SECURITY] if we are started as root, then here drop the UID/GID (we retain CAPs).
+	datastore::set_home(home_dir);
```

Comment:
I would move  this datastore::set_home() into  my_cap::drop_root()
because details how to switch user home, are responsibility of this function drop_root() that is galaxy42 internal anyway (and not an universal low-level function)